python/lsst/afw/image/exposure/_exposureContinued.py

```diff
@@ -94,6 +96,25 @@ def setVariance(self, variance):
 
     variance = property(getVariance, setVariance)
 
+    def getConvexPolygon(self):
+        """Get the convex polygon associated with the bounding box corners.
```

Comment:
The docs for this should probably note that this polygon is not a perfect sky-match to the detector boundaries. If people use this to check whether a sky point is within a detector, they need to also do a direct `skyToPixel` check afterwards, if they care about ~few pixel accuracy.

On that note: what's the rough value of the inaccuracy of this polygon? I'd expect a handful of pixels at most, and I hope that it's always strictly larger than the projected detector, but I don't have a good head for sky polygons.

Reply:
The inaccuracy depends on the size of the exposure and the latitude, I think.  If you need a cushion you can always dilate it, there are routines in `sphgeom`.

Reply:
I did some tests on HSC detectors in the center and the edge, and found that growing the bounding box by 7 pixels was sufficient to ensure that the polygon contained the entire detector.  Therefore, I set a default padding of 10 pixels.

One thing that I'm now unsure of is that this will be a larger/different polygon than is retrieved by the skymap `patch_info.outer_sky_polygon`, which doesn't need a buffer.

Reply:
I suppose there's no way for this to know whether it's operating on a patch or a single exposure? (I hear Jim saying "Yes we should have had different classes for those things!" in the background)

Reply:
What's hard is making this work idiomatically, but one could say "if this is a SIP-type WCS we need padding", but what could the parameter be called in this case?

Reply:
Please update the docs for this, per the above discussion.

Reply:
@parejkoj are you asking just to update the docstring with something like "The returned polygon has additional padding to ensure that the bounding box is entirely contained within it." or something else?

Reply:
Yes, that and something about the user needing to do a second check on the original pixel bbox, "is this sky point on the detector?" is the goal.

Reply:
```
        Get the convex polygon associated with the bounding box corners.

        The returned polygon has additional padding to ensure that the
        bounding box is entirely contained within it. To ensure a set of
        ra/dec points are entirely contained within a detector, a second
        check of points contained within the convex polygon must be
        projected to x/y and checked against the exposure bounding box.
```

Reply:
Second sentence, maybe be explicit about it?
```
...entirely contained within this exposure, run ``exposure.wcs.skyToPixel()`` to get the points in pixel coordinates, and check that those points are in the exposure's pixel box with ``exposure.getBBox().contains()``.
```